### PR TITLE
bots: Fix heartbeat in run-queue

### PR DESCRIPTION
**Pull request by Gundersanne**

### CODE REVIEW

File: bots/run-queue [hunk 1]
```diff
@@ -52,6 +53,7 @@ def main():
     connection = pika.BlockingConnection(pika.ConnectionParameters(
         host=host,
         port=int(port),
+        heartbeat_interval=60,
```
Comment (martinpitt) [inline bots/run-queue]: Do you want to leave this in the final commit? It seems a bit more practical (and less redundant) to me to set it in rabbitmq.conf on the deployment.

File: bots/run-queue [hunk 2]
```diff
@@ -84,7 +86,13 @@ def main():
         body = json.loads(body)
         sys.stderr.write("Consuming {0} task:\n{1}\n".format(body['type'],json.dumps(body, indent=2, sort_keys=True)))
         sys.stderr.flush()
-        if subprocess.call(body['command'], shell=True) in [0, 2]:
+        p = subprocess.Popen(body['command'], shell=True)
+        p.poll()
+        while p.returncode is None:
```
Comment (martinpitt) [inline bots/run-queue]: More elegant to do `while p.poll() is None`, and drop the two explicit calls.

**Comment by Gundersanne** [comment]
don't merge yet, wanna test some more first

File: bots/run-queue [hunk 3]
```diff
@@ -84,7 +85,11 @@ def main():
         body = json.loads(body)
         sys.stderr.write("Consuming {0} task:\n{1}\n".format(body['type'],json.dumps(body, indent=2, sort_keys=True)))
         sys.stderr.flush()
-        if subprocess.call(body['command'], shell=True) in [0, 2]:
+        p = subprocess.Popen(body['command'], shell=True)
+        while p.poll is None:
```
Comment (martinpitt) [inline bots/run-queue]: It's a function call `p.poll()` :)

**Pull request merged**